app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_post_user_counter
+    author.increment!(:posts_counter)
+  end
+
```

Comment:
- [ **OPTIONAL** ] I think it would be better if you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private. i.e.
```ruby
class Post < ApplicationRecord
  belongs_to :author, class_name: 'User'
  has_many :comments
  has_many :likes

  after_save :update_posts_counter

 def five_most_recent_comments
    comments.last(5)
  end

  private

   def update_post_user_counter
    author.increment!(:posts_counter)
  end
end